Fix link creation in search in document tool if using TYPO3 speaking urls (slug).

albig opened the pull request:
This would fix probably #747

beatrycze-volk committed:
Fix link creation for configured slug

beatrycze-volk committed:
Fix Codacy warnings

albig commented on Resources/Public/Javascript/PageView/SearchInDocument.js:
```diff
@@ -234,15 +256,22 @@ function getNavigationButtons(start, numFound) {
     }
 
+    if (!pageFound) {
+        var url = baseUrl.split('/');
```
Unfortunately, this code fails at this point: 

```
SearchInDocument.js?1640205096:271 Uncaught ReferenceError: baseUrl is not defined
    at getCurrentPage (SearchInDocument.js?1640205096:271)
    at addImageHighlight (SearchInDocument.js?1640205096:286)
    at Object.success (SearchInDocument.js?1640205096:366)
    at c (jquery-3.5.1.min.js?1605185398:2)
    at Object.fireWith [as resolveWith] (jquery-3.5.1.min.js?1605185398:2)
    at l (jquery-3.5.1.min.js?1605185398:2)
    at XMLHttpRequest.<anonymous> (jquery-3.5.1.min.js?1605185398:2)
```

albig replied:
If I add  `var baseUrl = getBaseUrl(" ");` above, it doesn't failt. But the highlighting isn't shown.

beatrycze-volk replied:
I have added logger for page. Could you tell me for which page highlighting is performed? (I have no test system with configured slugs, so I'm testing with use of fake links and there it looks correct) I was assuming that problem with highlight after link correction comes from incorrect page number taken by highlight method. This page is checked out by getCurrentPage() method.

albig reviewed (commented):
Please have a look at my comments. It doesn't work yet for me :-(

beatrycze-volk committed:
Add missing getBaseUrl

beatrycze-volk committed:
Add parse to int page value from array

It is probable that comparasion in getCurrentPage fails becase of page variable's type

albig commented on Resources/Public/Javascript/PageView/SearchInDocument.js:
```diff
@@ -233,16 +255,24 @@ function getNavigationButtons(start, numFound) {
+        var url = baseUrl.split('/');
+        page = parseInt(url.pop(), 10);
+    }
+    console.log('Page for highlight: ' + page);
```
```suggestion
```

albig reviewed (commented):
This works for me with and without Slug configuration.

Please just remove the console.log().

beatrycze-volk committed:
Update Resources/Public/Javascript/PageView/SearchInDocument.js

Co-authored-by: Alexander Bigga <alexander.bigga@slub-dresden.de>
